From 041dedfbfbf45ae38cb4959cacce1920f28ffbf3 Mon Sep 17 00:00:00 2001
From: James Slagle <jslagle@redhat.com>
Date: Tue, 9 Dec 2014 14:14:53 -0500
Subject: [PATCH] Allow for disabling rhel registration

When using the rhel element, it can be uesful to disable registration.
For instance, if using an internal mirror repository that does not
require registration/authentication.

Change-Id: I42fa34e7f146137e5b1b9c5d7785702311e1dca1
---
 .../os-refresh-config/pre-configure.d/06-rhel-registration           | 3 +++
 elements/rhel-common/pre-install.d/00-rhel-registration              | 5 ++++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/elements/rhel-common/os-refresh-config/pre-configure.d/06-rhel-registration b/elements/rhel-common/os-refresh-config/pre-configure.d/06-rhel-registration
index f85569f..d7fd5be 100755
--- a/elements/rhel-common/os-refresh-config/pre-configure.d/06-rhel-registration
+++ b/elements/rhel-common/os-refresh-config/pre-configure.d/06-rhel-registration
@@ -113,6 +113,9 @@ case "${REG_METHOD:-}" in
         # beta-rpms repo only needed to support the katello-ca rpm above.
         subscription-manager repos --disable ${satellite_repo}
         ;;
+    disable)
+        echo "Disabling RHEL registration"
+        ;;
     *)
         echo "WARNING: only 'portal' and 'satellite' are valid values for REG_METHOD."
         exit 0
diff --git a/elements/rhel-common/pre-install.d/00-rhel-registration b/elements/rhel-common/pre-install.d/00-rhel-registration
index ced9529..0314019 100755
--- a/elements/rhel-common/pre-install.d/00-rhel-registration
+++ b/elements/rhel-common/pre-install.d/00-rhel-registration
@@ -96,7 +96,10 @@ case "${REG_METHOD:-}" in
         # beta-rpms repo only needed to support the katello-ca rpm above.
         subscription-manager repos --disable rhel-7-server-rh-common-beta-rpms
         ;;
+    disable)
+        echo "Disabling RHEL registration"
+        ;;
     *)
-        echo "WARNING: only 'portal' and 'satellite' are valid values for REG_METHOD."
+        echo "WARNING: only 'portal', 'satellite', and 'disable' are valid values for REG_METHOD."
         exit 1
 esac
